hw/mcu/microchip/pic32mz2048efg100/src/hal_i2c.c

```diff
@@ -0,0 +1,276 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+#include <xc.h>
+#include "bsp/bsp.h"
+#include <hal/hal_gpio.h>
+#include <hal/hal_i2c.h>
+#include <mcu/p32mz2048efg100.h>
+#include "mcu/mips_hal.h"
+#include "syscfg/syscfg.h"
```

Comment:
You probably want to move include for syscfg.h higher. I recommend including it before any of the other Mynewt headers.